python/pyiceberg/table/partitioning.py

```diff
@@ -57,24 +81,33 @@ class PartitionSpec:
         last_assigned_field_id(int): auto-increment partition field id starting from PARTITION_DATA_ID_START
     """
 
-    schema: Schema
-    spec_id: int
-    fields: Tuple[PartitionField, ...]
-    last_assigned_field_id: int
-    source_id_to_fields_map: Dict[int, List[PartitionField]] = field(init=False, repr=False)
-
-    def __post_init__(self):
-        source_id_to_fields_map = {}
-        for partition_field in self.fields:
-            source_column = self.schema.find_column_name(partition_field.source_id)
-            if not source_column:
-                raise ValueError(f"Cannot find source column: {partition_field.source_id}")
-            existing = source_id_to_fields_map.get(partition_field.source_id, [])
-            existing.append(partition_field)
-            source_id_to_fields_map[partition_field.source_id] = existing
-        object.__setattr__(self, "source_id_to_fields_map", source_id_to_fields_map)
-
-    def __eq__(self, other):
+    spec_id: int = Field(alias="spec-id")
+    fields: Tuple[PartitionField, ...] = Field()
+    last_assigned_field_id: int = Field(alias="last-assigned-field-id", default=_PARTITION_DATA_ID_START)
```

Comment:
We probably don't need this. It is just a high watermark of the field IDs in this spec. We can just define a property that returns it:

```python
    @property
    def last_assigned_field_id(self):
        return max(pf.field_id for pf in self.fields)
```